hnn_core/dipole.py

```diff
@@ -137,9 +137,27 @@ def simulate_dipole(net, n_trials=1, n_jobs=1):
     dpl: list | instance of Dipole
         The dipole object or list of dipole objects if n_trials > 1
     """
-    parallel, myfunc = _parallel_func(_clone_and_simulate, n_jobs=n_jobs)
-    out = parallel(myfunc(net.params, idx) for idx in range(n_trials))
-    dpl, spiketimes, spikegids = zip(*out)
+
+    from .parallel import create_parallel_context, get_nhosts
+
+    if n_jobs > 1:
```

Comment:
Could we somehow avoid two code paths for `n_jobs`? That's the beauty of joblibs as it ensures that what works on one core will work also on multiple cores

Reply:
The reason there are two code paths is to separate MPI ParallelContext from Joblibs. Nested parallelism will not even work, so we test for it with n_jobs. The same thing can be said of MPI, it will work on any number of cores. Except MPI and joblibs is not a good idea.

Reply:
@jasmainak this is the best option I have

Reply:
no longer needed